### ISSUE TITLE
Error when executing simulate command of the TTN Cli tool

### ISSUE THREAD

**Issue by cheribhai** [TheThingsNetwork/lorawan-stack#2178]
#### Summary
Simulate command of the TTN Cli tool gives an error

#### Steps to Reproduce
The following script creates an application, gateway, end-device and then runs the uplink simulate command

```
#!/bin/bash

GTW_ID=gtw-sim-2178-3
APP_ID=app-sim-2178-3
DEV_ID=dev-sim-2178
FREQ_PLAN=EU_863_870
LORAWAN_VERSION=1.0.3
LORAWAN_PHY_VERSION=1.0.3-a
APP_S_KEY=A0CAD5A30036DBE03096EB67CA975BAA
NWK_S_KEY=B7F3E161BC9D4388E6C788A0C547F255
DEV_ADDR=00E4304D
LINK_API_KEY=NNSXS.2ZDDRJK7TCFWWWC3KL6PSQYJIJKRHWD3RQMPFRA.5OTQHZM4F7VI4G2G7L4M3M2OEMVY4VSVLV4YB2PMDE2WCWVFVSOA

ttn-lw-cli gateways create $GTW_ID \
  --user-id $1 \
  --frequency-plan-id $FREQ_PLAN \
  --enforce-duty-cycle

ttn-lw-cli applications create $APP_ID --user-id $1

# ttn-lw-cli applications api-keys create \
#   --name link \
#   --application-id $APP_ID \
#   --right-application-link

ttn-lw-cli applications link set $APP_ID --api-key $LINK_API_KEY

ttn-lw-cli end-devices create $APP_ID $DEV_ID \
  --frequency-plan-id $FREQ_PLAN \
  --lorawan-version $LORAWAN_VERSION \
  --lorawan-phy-version $LORAWAN_PHY_VERSION \
  --abp \
  --session.dev-addr $DEV_ID \
  --session.keys.app-s-key.key $APP_S_KEY \
  --session.keys.nwk-s-key.key $NWK_S_KEY

ttn-lw-cli simulate uplink \
    --dev-addr $DEV_ADDR \
    --frm-payload 4869 \
    --lorawan-version $LORAWAN_VERSION \
    --log.level debug \
    --gateway-id $GTW_ID \
    --f-nwk-s-int-key $NWK_S_KEY \
    --app-s-key $APP_S_KEY \
    --lorawan-phy-version $LORAWAN_PHY_VERSION \
    --band-id $FREQ_PLAN \
```

#### What do you see now?
Command returns an error and no message is received on the TTN Device Data console ...
DEBUG Using access token (valid until 10:39AM)
DEBUG ccResolverWrapper: sending update to cc: {[{ourdomain.eu1.cloud.thethings.industries:8884  <nil> 0 <nil>}] <nil> <nil>}
DEBUG pickfirstBalancer: HandleSubConnStateChange: 0xc0002a8260, {CONNECTING <nil>}
DEBUG pickfirstBalancer: HandleSubConnStateChange: 0xc0002a8260, {READY <nil>}
INFO Sent uplink                            
DEBUG Finished streaming call                  duration=20.000540346s error=context deadline exceeded grpc_method=LinkGateway grpc_service=ttn.lorawan.v3.GtwGs namespace=grpc
error:unknown:unknown (context deadline exceeded)

#### What do you want to see instead?
Command executes as expected and the message is seen in the TTn Device Data console.

#### Environment
: ttn-lw-cli version                
The Things Network Command-line Interface: ttn-lw-cli
Version:             3.6.1
Build date:          2020-03-13T11:26:42Z
Git commit:          8be68232e
Go version:          go1.13.8
OS/Arch:             linux/amd64

**Comment by johanstokking** [comment]
@cheribhai please provide detailed steps to reproduce.

- how did you create the gateway?
- how did you create the device?
- what exactly is the command for simulate?

Note that the `simulate` command is hidden because it's highly experimental.

**Comment by cheribhai** [comment]
@johanstokking I have created the device (via ABP) and the gateway with TTN generated device address and keys.
The device and gateway have been created purely for simulation purpose and are not connected to any physical device
I have provided the ttn simulate command in the description (minus sensitive info)

**Comment by johanstokking** [comment]
OK, but please copy the exact commands so we can reproduce the issue.

Also, what do the server logs say, with debug logs enabled?

**Comment by cheribhai** [comment]
@johanstokking I have updated the 'Steps to reproduce' section with a bash script which contains all the details and commands to create the application, gateway, device and then run the simulate command.
We are not self-hosting the TTN V3 stack. We have (last week) subscribed to the TTN hosted V3 (Industries) stack, so I'm not really sure If I can get to the server logs myself.
I do need to run some end to end tests before migrating the applications / devices from V2 to V3, so if there is any other option to simulate (maybe direct api calls) please advise.

**Comment by johanstokking** [comment]
OK I see.

Please specify frequency, bandwidth and spreading factor. Since you didn't specify the frequency plan as part of the device create, only the three default channels are enabled (so `868100000`, `868300000` and `868500000`), bandwidth is `125000` and spreading factor like `7`. You also need to specify `f-cnt` (incrementing).

Finally, you can subscribe to gateway events (`ttn-lw-cli events subscribe --gateway-id $GTW_ID`)

**Comment by cheribhai** [comment]
As can be seen in the description, I am specifying the frequency plan in the request.
I can confirm this with the call
t```
tn-lw-cli end-devices get $APP_ID $DEV_ID --frequency-plan-id
{
  "ids": {
    "device_id": "dev-sim-2178",
    "application_ids": {
      "application_id": "app-sim-2178-3"
    },
    "dev_addr": "00E4304D"
  },
  "created_at": "2020-03-20T17:46:14.357Z",
  "updated_at": "2020-03-20T17:46:45.514700632Z",
  "network_server_address": "<domain>.cloud.thethings.industries",
  "frequency_plan_id": "EU_863_870"
}

```
I had made a mistake with,
`--session.dev-addr $DEV_ID \`
which I have corrected to 
`--session.dev-addr $DEV_ADDR \`

Running the simulate uplink with explicitly specifying the frequency, bandwith and f-cnt still gives the same error,
```
ttn-lw-cli simulate uplink \
    --dev-addr $DEV_ADDR \
    --frm-payload 4869 \
    --lorawan-version $LORAWAN_VERSION \
    --log.level debug \
    --gateway-id $GTW_ID \
    --f-nwk-s-int-key $NWK_S_KEY \
    --app-s-key $APP_S_KEY \
    --lorawan-phy-version $LORAWAN_PHY_VERSION \
    --band-id $FREQ_PLAN \
    --bandwidth 125000 \
    --frequency 868500000 \
    --spreading-factor 7 \
    --f-cnt 1
```

**Comment by cheribhai** [comment]
The gateway events show the error ..

```
  "data": {
    "@type": "type.googleapis.com/ttn.lorawan.v3.ErrorDetails",
    "namespace": "pkg/gatewayserver",
    "name": "host_handle",
    "message_format": "host `{host}` failed to handle message",
    "attributes": {
        "host": "cluster"
      },
    "cause": {
      "namespace": "pkg/networkserver",
      "name": "device_not_found",
      "message_format": "device not found",
      "correlation_id": "b643b1da80814b4fa81b438625abd73c",
      "code": 5
    },
    "code": 5
  }
```

**Comment by johanstokking** [comment]
> As can be seen in the description, I am specifying the frequency plan in the create device request.

Right, that wasn't clear on my side. I meant that with ABP, you didn't specify the enabled channels. Those are the "factory reset frequencies". If you don't specify those, only 868.1, 868.3 and 868.5 are enabled.

In any case, you can either specify `frequency` or `channel-index`. If you don't specify either, `channel-index` is 0, and 868.1 as channel 0 is used. Also, you can either specify `data-rate-index` or `spreading-factor`/`bandwidth`. If you don't specify either, data rate 0 is used, which is SF12BW125. So that's not the issue.

I cannot reproduce your issue locally with TTS. It works immediately. What I had to fix though, is to use `$DEV_ADDR` instead of `$DEV_ID` in your example.

Since the NS cannot find the device, I suppose the problem is the (reused) frame counter. Please make sure you increase that counter on any attempt. You may want to set `--mac.settings.resets-f-cnt` to `true` for testing.

What we have currently running in The Things Industries Cloud Hosted is 3.6.1. We patched some uplink related issues in 3.7.0-rc1, which we're deploying Monday.

**Comment by cheribhai** [comment]
@johanstokking Incrementing the `f-cnt` counter was what I was missing (my bad). Sorry about that - I should have seen this.

But thanks a ton! It works now and is really helpful - I will be setting `--mac.settings.resets-f-cnt` to `true` for devices that will only be used for simulation.

I see another (random) error, but I have opened a separate ticket for this - https://github.com/TheThingsNetwork/lorawan-stack/issues/2210
